backends/cxxrtl/cxxrtl.h

```diff
@@ -439,6 +440,27 @@ struct value : public expr_base<value<Bits>> {
 		return value<Bits> { 0u }.sub(*this);
 	}
 
+	template<size_t ResultBits>
+	value<ResultBits> mul(const value<Bits> &other) const {
+		value<ResultBits> result;
+
+		wide_chunk_t wide_result[result.chunks + 1] = {};
+
+		for (size_t n = 0; n < chunks; n++) {
+			for (size_t m = 0; m < chunks && n + m < result.chunks; m++) {
+				wide_result[n + m] += wide_chunk_t(data[n]) * wide_chunk_t(other.data[m]);
+				wide_result[n + m + 1] += wide_result[n + m] >> chunk::bits;
+				wide_result[n + m] &= chunk::mask;
+			}
+		}
+
+		for (size_t n = 0; n < result.chunks; n++) {
+			result.data[n] = wide_result[n];
+		}
+		result.data[result.chunks - 1] &= result.msb_mask;
+		return result;
+	}
+
```

Comment:
I'd probably put `mul` below the `*cmp` to group similar operations together, but this isn't important.